docs/docs/dev_docs/contracts/syntax/slow_updates_tree.md

```diff
@@ -0,0 +1,268 @@
+    INT_UAP[update_at_public] <--> STC_UAP[update_at_public]
+    INT_UA[update_at_private] <--> STC_UA[update_at_private]
+
+    STC_RA <--> VMP[verify_membership_proof]
+    STC_UA <--> CR[compute_roots]
+    
+    subgraph INT[Interface]
+        INT_RAP
+        INT_UAP
+        INT_RA
+        INT_UA
+    end
+    
+    subgraph STC[SlowTree.nr]
+        STC_RAP
+        STC_UAP
+        STC_RA
+        STC_UA
+    end
+    
+    subgraph SMT[SlowMap Type]
+        Change{Epoch Over} -->|True| Current{Current}
+        Change -->|False| Pending{Pending}
+        Current --> Current1[Current Commitment 1]
+        Current --> CurrentM[Current Commitment M]
+        CurrentM --> Value1[Current Value 1]
+        CurrentM --> Value2[Current Value 2]
+        CurrentM --> ValueN[Current Value N]
+        Pending --> PendingM[Pending Commitment 1]
+        PendingM --> PValue1[Pending Value 1]
+        PendingM --> PValue2[Pending Value 2]
+        PendingM --> PValueN[Pending Value N]
+    end
+
+    style INT fill:#fff,stroke:#333,stroke-width:1px
+    style STC fill:#fff,stroke:#333,stroke-width:1px
+    style SMT fill:#fff,stroke:#333,stroke-width:1px
+```
+
+# How to integrate a slow updates tree
+
+1. Copy the *SlowTree.nr* example and its dependencies, found [here](https://github.com/AztecProtocol/aztec-packages/tree/master/yarn-project/noir-contracts/src/contracts/slow_tree_contract). Replace the constants with whatever you like and deploy it to your sandbox
+2. Copy the *SlowMap interface* for easy interaction with your deployed SlowTree. Find it [here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/interfaces.nr)
+3. Import this interface into your contract
+
+#include_code interface yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+5. Store a slow updates tree in both public and private storage
+
+#include_code slow_updates_storage yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+6. Store the SlowTree address in private storage as a FieldNote
+
+#include_code constructor yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+7. Store the SlowTree address in public storage and initialize an instance of SlowMap using this address
+
+#include_code write_slow_update_public yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+8. Now you can read and update from private functions:
+
+#include_code get_and_update_private yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+9. Or from public functions:
+
+#include_code get_public yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+View the [reference](slow_updates_tree.md#reference) for more information.
+
+## Exploring an example integration through a **`TokenBlacklist`** Smart Contract
```

Comment:
isn't this section an exact copy paste of the previous section? should we just remove it?
Your previous section just seems very very nice

Reply:
i think its worth it to keep it in. examples are always really good. it will help understand what the previous how-to section actually means